aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -23,6 +23,20 @@ void bitwise_not_kernel_cuda(TensorIterator& iter) {
   }
 }
 
-REGISTER_DISPATCH(bitwise_not_stub, &bitwise_not_kernel_cuda);
 
+void sign_kernel_cuda(TensorIterator& iter){
+    if (iter.dtype() == ScalarType::Bool) {
+        // sign over Bool tensor is defined as identity function https://github.com/pytorch/pytorch/pull/22861#discussion_r304126678
```

Comment:
I don't think this is the correct way for non-in-place version. I think you can do this in `sign_out` by cloning the input Tensor when the type is bool.

Reply:
Yeah, this isn't correct since it leaves the output unchanged (with potentially uninitialized memory).

You should be able to use the same code for bool as for other types. Does it not work?

Reply:
Thanks for your comments, I pushed a little bit too early and effectively it leaves memory uninitialised.

The idea was to avoid having to iterate over the whole tensor to generate a perfect copy (identity).

Reply:
@mfuntowicz I would prefer, if possible, to use the sample implementation for bool as other types since it minimizes the chances for mistakes

Reply:
@colesbury But if a simple cloning is directly handled in `sign_out`, won't it be more efficient?

Reply:
@xuhdev it's going to be efficient enough. Efficiency of torch.sign bool data types is not a significant concern. It's implemented on bools not because it's useful in itself but because it's convenient to have functions be generic in their input types.

Reply:
Argh, I pushed the copy_ version, let me know which direction is preferred, I'll move the right way.

Reply:
Please use the sample implementation for bool. In case you're not aware there is an AT_DISPATCH_ALL_TYPES_AND2 which you can call like:

https://github.com/pytorch/pytorch/blob/14e8fb70a138c13df75fd2a0ef72503b3591cc6e/aten/src/ATen/native/cuda/FillKernel.cu#L10

Reply:
Might be good now.